File: Problems/0001_0099/0061_Rotate_List/Project_Java/OperateListNode.java
public class OperateListNode {
    public ListNode createListNode(String flds) {
        if (flds.length() == 0)
            return null;
        return createSubListNode(flds.split(","), 0);
    }

    public ListNode createSubListNode(String[] flds, int index) {
        if (flds == null)
            return null;

        if (index >= flds.length)
            return null;
        
        ListNode node = new ListNode(Integer.parseInt(flds[index]));
        node.next = createSubListNode(flds, index + 1);

        return node;
    }

    public String listNodeToString(ListNode node) {
        if (node == null)
            return "";

        String resultStr = Integer.toString(node.val);

        if (node.next != null)
            resultStr += " -> " + listNodeToString(node.next);

        return (resultStr);
    }
}

File: Problems/0400_0499/0445_Add_Two_Numbers_II/Project_Go/solution/OperateListNode.go
package solution

import "strconv"

func CreateListNode(flds string) *ListNode {
	if len(flds) <= 0 {
		return nil
	}
	return createSubListNode(StringToIntArray(flds), 0)
}

func createSubListNode(nums []int, index int) *ListNode {
	if nums == nil {
		return nil
	}

	if index >= len(nums) {
		return nil
	}

	node := new(ListNode)
	node.Val = nums[index]
	node.Next = createSubListNode(nums, index+1)

	return node
}

func ListNodeToString(node *ListNode) string {
	if node == nil {
		return ""
	}

	resultStr := strconv.Itoa(node.Val)

	if node.Next != nil {
		resultStr += " -> " + ListNodeToString(node.Next)
	}

	return (resultStr)
}
